Fabric.Authorization.Domain/Resolvers/Permissions/RolePermissionResolverService.cs

```diff
@@ -78,14 +78,16 @@ public async Task<PermissionResolutionResult> Resolve(PermissionResolutionReques
             };
         }
 
-        private static bool BypassRoleEvaluation(Role role, IReadOnlyCollection<string> groupNames, PermissionResolutionRequest permissionResolutionRequest)
+        private static bool BypassRoleEvaluation(Role role, IReadOnlyCollection<Group> groups, PermissionResolutionRequest permissionResolutionRequest)
         {
-            return !(role.Groups.Any(g => groupNames.Contains(g.Name, StringComparer.OrdinalIgnoreCase)) || role.Users.Any(
-                         u => u.SubjectId.Equals(permissionResolutionRequest.SubjectId,
-                                  StringComparison.OrdinalIgnoreCase) &&
-                              u.IdentityProvider.Equals(permissionResolutionRequest.IdentityProvider,
-                                  StringComparison.OrdinalIgnoreCase))) || role.IsDeleted ||
-                   role.Permissions == null || role.Permissions.Count == 0;
+            return !(role.Groups.Any(g => groups.Select(g1 => g1.GroupIdentifier).Contains(g.GroupIdentifier, new GroupIdentifierComparer()))
```

Comment:
is this where you think we need to use the *correct* ID, either from Azure or other?

Reply:
The GroupIdentifier is a property consisting of IdP + tenant ID + group name. This logic was switched from comparing group names to comparing `GroupIdentifier` objects.

Reply:
The Azure identifier is used to correlate internal Fabric.Auth groups to Azure groups. We correlate through the new `ExternalIdentifier` column. Access tokens from Azure AD will have that in the `groups` claim so we have to translate internally. This is done up the call stack from this method.